tools/editors-libraries.adoc

```diff
@@ -0,0 +1,177 @@
+* `close(): void`: Closes the Editor.
+* `envelopeApi: MessageBusClientApi<KogitoEditorEnvelopeApi>`: Advanced Editor API.
+* `canvas`: Canvas API that exposes methods to manipulate the canvas.
+ ** `getNodeIds(): Promise<string[]>`: Returns a Promise containing the ID attributes of all nodes displayed in the editor canvas.
+ ** `getBackgroundColor(uuid: string): Promise<string>`: Returns a Promise containing the background color of a node with the provided UUID.
+ ** `setBackgroundColor(uuid: string, backgroundColor: string): Promise<void>`: Sets the background color of a node with the provided UUID.
+ ** `getBorderColor(uuid: string): Promise<string>`: Returns a Promise containing the border color of a node with the provided UUID.
+ ** `setBorderColor(uuid: string, borderColor: string): Promise<void>`: Sets the border color of a node with provided UUID.
+ ** `getLocation(uuid: string): Promise<number[]>`: Returns a Promise containing the canvas location of a node with the provided UUID.
+ ** `getAbsoluteLocation(uuid: string): Promise<number[]>`: Returns a Promise containing the window location for a node with the provided UUID.
+ ** `getDimensions(uuid: string): Promise<number[]>`: Returns a Promise containing the dimensions of a node with provided UUID.
+ ** `applyState(uuid: string, state: string): Promise<void>`: Applies state to a node given the UUID [None, Selected, Highlight, Invalid].
+ ** `centerNode(uuid: string): Promise<void>`: Centers node on viewable Canvas.
+
+== New DMN Editor
+
+The modern DMN Editor, built with React, is faster and supports up to DMN 1.5 specifications while maintaining the same simple-to-use API.
+
+.New DMN Editor
+image::tools/editorsLibraries/new-dmn-editor.png[]
+
+=== Installing and using the new DMN Editor
+
+First, add the library to the project's `package.json`:
+[source,shell,subs="+attributes"]
+----
+npm install @ibm/@ibm/bamoe-standalone-dmn-editor@{NPM_VERSION}
+----
+
+Import the editor library to a JS file with:
+[source,javascript,subs="+attributes"]
+----
+import * as NewDmnEditor from "@ibm/bamoe-standalone-dmn-editor/dist"
+----
+
+In the HTML, add a `div` container for each Editor; that's where the library will load an iFrame with all the Editor's resources:
+
+[source,html,subs="+attributes"]
+----
+<div id="new-dmn-editor-container"/>
+----
+
+Then load the Editor, passing the reference to the `div` container:
+
+[source,javascript,subs="+attributes"]
+----
+const newDmnEditor = NewDmnEditor.open({
+ container: document.getElementById("new-dmn-editor-container"), <1>
+ initialFileNormalizedPosixPathRelativeToTheWorkspaceRoot: "model.dmn" <2>
+ initialContent: Promise.resolve(""), <3>
+ readOnly: false, <4>
+ origin: "*", <5>
+ onError: () => {}, <6>
+ resources: new Map([]), <7>
+});
+----
+<1> `container`: HTML element to which the Editor will be appended to.
+<2> `initialContent`: Promise to the XML content of a DMN file for the DMN Editor. Can be empty. Examples:
```

Comment:
Promise to the XML content of a DMN file for the DMN editor and it can be empty. Examples: